### ISSUE TITLE
[REQUEST] Table row style should overwrite table style

### ISSUE THREAD

**Issue by ewels** [willmcgugan/rich#1129]
**Describe the bug**
When making a table, setting the `style` on a row does nothing if it's been set on the `table`.

Setting colours in-line with colour strings does work as expected.

**To Reproduce**

```python
from rich.table import Table
from rich import print

table = Table(style="green")
table.add_column("RESULTS SUMMARY")
table.add_row("[✔] 3 Tests Passed", style="green")
table.add_row("[1] 5 Test Warnings", style="yellow")
table.add_row("[✗] 9 Tests Failed", style="red")
print(table)

table = Table(style="green")
table.add_column("RESULTS SUMMARY")
table.add_row("[green][✔] 3 Tests Passed")
table.add_row("[yellow][1] 5 Test Warnings")
table.add_row("[red][✗] 9 Tests Failed")
print(table)
```

<img src="https://user-images.githubusercontent.com/465550/112146178-90e5a580-8bdb-11eb-9f90-f1f878665446.png" height=300>

**Comment by willmcgugan** [comment]
Should be fixed in v10.0.0

**Comment by ewels** [comment]
Fantastic, thanks!
